### PR TITLE
feature/my open position

### PR DESCRIPTION
**Pull request by JGiter**
- Added My Orders tab with My Asks and My Bids tables
- Tables rows are filtered by generation time period, device type and facility name

### CODE REVIEW

File: packages/exchange/src/pods/order/product.dto.ts [hunk 1]
```diff
@@ -38,6 +38,9 @@ export class ProductDTO {
     @Validate(GridOperatorValidator)
     public gridOperator?: string[];
 
+    @IsOptional()
+    public externalDeviceId?: { id: string; type: string };
```
Comment (josipbagaric) [inline packages/exchange/src/pods/order/product.dto.ts]: Replace with the proper type `IExternalDeviceId`.

https://github.com/energywebfoundation/origin/blob/52e5fccdeb303133b3db9413fc317e4a1c60ebca/packages/origin-backend-core/src/Device.ts#L10

File: packages/origin-ui-core/src/components/Modal/AskDetailslModal.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+interface IOwnProps {
+    ask: Order;
+    close: () => void;
+}
+
+export const AskDetailsModal = (props: IOwnProps) => {
+    const { ask, close } = props;
+    return (
+        <Dialog open={ask !== null} onClose={close}>
+            View ask details modal
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/Modal/AskDetailslModal.tsx:14]: Needs to be localized.

File: packages/origin-ui-core/src/components/Modal/BidDetalisModal.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+interface IOwnProps {
+    bid: Order;
+    close: () => void;
+}
+
+export const BidsDetailsModal = (props: IOwnProps) => {
+    const { bid, close } = props;
+    return (
+        <Dialog open={bid !== null} onClose={close}>
+            View ask details modal
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/Modal/BidDetalisModal.tsx:14]: Localize

File: packages/origin-ui-core/src/components/Modal/RemoveOrderConfirmation.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+interface IOwnProps {
+    order: Order;
+    close: () => void;
+}
+
+export const RemoveOrderConfirmation = (props: IOwnProps) => {
+    const { order, close } = props;
+    return (
+        <Dialog open={order !== null} onClose={close}>
+            View ask details modal
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/Modal/RemoveOrderConfirmation.tsx:14]: Localize

File: packages/origin-ui-core/src/components/Table/PaginatedLoaderHooks.tsx [hunk 5]
```diff
@@ -205,10 +206,13 @@ export function checkRecordPassesFilters(
...
     for (const filter of filters) {
         const filteredPropertyResolvedValue = filter.property(record);
-
         if (typeof filteredPropertyResolvedValue !== 'undefined') {
             switch (filter.input.type) {
                 case CustomFilterInputType.string:
+                    console.log(
+                        '>>> String filter: record to filter:',
+                        filteredPropertyResolvedValue
+                    );
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/Table/PaginatedLoaderHooks.tsx]: Do we need to keep these console logs?

File: packages/origin-ui-core/src/components/orders/AsksTable.tsx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    const removeAsk = (rowIndex: number) => {
+        const { askId } = rows[rowIndex];
+        const ask = asks.find((o) => o.id === askId);
+        setToRemove(ask);
+    };
+
+    const actions = [
+        {
+            icon: <Visibility />,
+            name: 'View',
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/orders/AsksTable.tsx]: Localize

File: packages/origin-ui-core/src/components/orders/AsksTable.tsx [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+            onClick: (row: string) => viewDetails(parseInt(row, 10))
+        },
+        {
+            icon: <Remove />,
+            name: 'Remove',
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/orders/AsksTable.tsx]: Localize

File: packages/origin-ui-core/src/components/orders/BidsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+    const removeBid = (rowIndex: number) => {
+        const { bidId } = rows[rowIndex];
+        const bid = bids.find((o) => o.id === bidId);
+        setToRemove(bid);
+    };
+
+    const actions = [
+        {
+            icon: <Visibility />,
+            name: 'View',
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/orders/BidsTable.tsx]: Localize?

File: packages/origin-ui-core/src/components/orders/BidsTable.tsx [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+            onClick: (row: string) => viewDetails(parseInt(row, 10))
+        },
+        {
+            icon: <Remove />,
+            name: 'Remove',
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/orders/BidsTable.tsx]: Localize

File: packages/origin-ui-core/src/styles/variables.scss [hunk 10]
```diff
@@ -21,6 +21,8 @@ $createBundleCertificatesGroupBg: #3b3b3b;
 $createBundleCertificatesGroupHeaderBg: #373737;
 $createBundleCertificateForSale: #303030;
 $createBundleSelectedCertificate: #3a2146;
+$filterBg: #3a3a3a;
+$filterItemBg: #2d2d2d;
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/styles/variables.scss]: **Do we really need these many new colors**? Can't we replace most of these colors with a few colors which can be reused?

1. Are `$filterBg` and `$createBundleCertificatesGroupBg` really that different that they have to be defined differently?
2. `$bundleDetailsBg` and `$bundleCardBorderBg` are basically the **same color**, why are they defined twice?
3. `$filterItemBg` and `$createBundleCertificatesBg` are the **same color**.

Can't these just be replaced with one color called `$secondaryColor: #3b3b3b`?

Then we can create different colors such as `$secondaryColorDarker: darken($secondaryColor, 10%)`.

Adding a new color for each new component that we build is stupid. We should have a set number of colors in the whole application, which we then reuse, and that can be easily customized.

I suggest a similar approach to [Boostrap](https://getbootstrap.com/docs/4.0/utilities/colors/#background-color), where we can define a **primary** (e.g. purple) and **secondary** color (e.g. grey), and then all other variations of that color is dynamically generated using `darken()` and `lighten()` variations of the primary and secondary colors.

Does this make sense?

File: packages/origin-ui-core/src/utils/exchange/types.ts [hunk 11]
```diff
@@ -13,6 +13,7 @@ export interface IProductDTO {
     generationFrom?: string;
     generationTo?: string;
     gridOperator?: string[];
+    externalDeviceId?: { id: string; type: string };
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/utils/exchange/types.ts]: Replace with the proper type `IExternalDeviceId`.

https://github.com/energywebfoundation/origin/blob/52e5fccdeb303133b3db9413fc317e4a1c60ebca/packages/origin-backend-core/src/Device.ts#L10

**Review by josipbagaric — APPROVED** [review]

**Review by josipbagaric — CHANGES_REQUESTED** [review]
Please check the review for necessary changes.

File: packages/origin-ui-core/src/components/Modal/AskDetailslModal.tsx (hunk 2 above)
Comment (JGiter) [reply packages/origin-ui-core/src/components/Modal/AskDetailslModal.tsx:14]: These modals (Ask and Bids details and remove) are templates and developed in other PR

File: packages/origin-ui-core/src/components/Table/PaginatedLoaderHooks.tsx (hunk 5 above)
Comment (JGiter) [reply packages/origin-ui-core/src/components/Table/PaginatedLoaderHooks.tsx]: sorry. I am just keep to forget to remove consols. May be we should add eslint rule which forbids them

File: packages/origin-ui-core/src/styles/variables.scss (hunk 10 above)
Comment (JGiter) [reply packages/origin-ui-core/src/styles/variables.scss]: Agree. I'll try to implement this approach

File: packages/origin-ui-core/src/components/Modal/BidDetalisModal.tsx (hunk 3 above)
Comment (JGiter) [reply packages/origin-ui-core/src/components/Modal/BidDetalisModal.tsx:14]: Its temporary templates

File: packages/origin-ui-core/src/components/Modal/RemoveOrderConfirmation.tsx (hunk 4 above)
Comment (JGiter) [reply packages/origin-ui-core/src/components/Modal/RemoveOrderConfirmation.tsx:14]: Temporary templates

File: packages/origin-ui-core/src/components/Table/PaginatedLoaderHooks.tsx (hunk 5 above)
Comment (josipbagaric) [reply packages/origin-ui-core/src/components/Table/PaginatedLoaderHooks.tsx]: We thought about it, but we do have intentional console.logs in the codebase, so it wouldn't work.

**Review by josipbagaric — APPROVED** [review]

**Pull request merged**